### PR TITLE
Date header improvements

### LINKED ISSUE #669: Date Header conformance
It seems the Date Header does not conform strictly too [RFC2616] https://www.ietf.org/rfc/rfc2616.txt, which states that dates should use the `GMT` timezone unlike http4s which uses `UTC`:

```
 All HTTP date/time stamps MUST be represented in Greenwich Mean Time
 (GMT), without exception. For the purposes of HTTP, GMT is exactly
 equal to UTC (Coordinated Universal Time).
```

### PR DESCRIPTION
**Pull request by cquiroz**
This PR brings some improvements to the Date header:

* It fixes #669 writing the Date header always using GMT
* It improves the parser of dates for the extra formats indicated in RFC 2616:
```
   HTTP applications have historically allowed three different formats
   for the representation of date/time stamps:

      Sun, 06 Nov 1994 08:49:37 GMT  ; RFC 822, updated by RFC 1123
      Sunday, 06-Nov-94 08:49:37 GMT ; RFC 850, obsoleted by RFC 1036
      Sun Nov  6 08:49:37 1994       ; ANSI C's asctime() format
```
* It contains some small fixes to code found while doing the PR, essentially some typos and unnecessary imports

### CODE REVIEW

**Comment by bryce-anderson** [comment]
Thanks for the PR! Looks good to me. I'll let it sit for further review for a day or so before merging. That will also give me time to look at it on something other then me phone! 😀

This looks like it could also target a bug fix branch.

File: core/src/main/scala/org/http4s/parser/AdditionalRules.scala [hunk 1]
```diff
@@ -79,13 +83,13 @@ private[parser] trait AdditionalRules extends Rfc2616BasicRules { this: Parser =
...
-                                   ("Monday"   ~ push(1)) |
-                                   ("Tuesday"  ~ push(2)) |
-                                   ("Wedsday"  ~ push(3)) |
-                                   ("Thursday" ~ push(4)) |
-                                   ("Friday"   ~ push(5)) |
-                                   ("Saturday" ~ push(6)) }
+  def Weekday: Rule1[Int] = rule { ("Sunday"    ~ push(0)) |
+                                   ("Monday"    ~ push(1)) |
+                                   ("Tuesday"   ~ push(2)) |
+                                   ("Wednesday" ~ push(3)) |
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/parser/AdditionalRules.scala]: Ouch.  How did that last all this time?!

**Comment by rossabaker** [comment]
Thanks, @cquiroz.  Excellent work.  I agree with @bryce-anderson on the targeting.  I'll make it happen.

**Comment by rossabaker** [comment]
Cherry-picked.  It will be in v0.14.3 and beyond.  Thanks!

**Pull request closed without merging**